### PR TITLE
 Documentation Gap Analysis

### PR DESCRIPTION
**Pull request by i-akruti**
Here is my analysis of the gaps in the Ultralytics documentation, highlighting some of the major pain points. Find out how I identify areas where users encounter friction, propose solutions, and explain the expected impact on usability and support.

## Problem Identification Process
The gap identification followed a combination of methods:

1. **User Journey Analysis:**  
   - Followed the “getting started” path as a new user with YOLOv11.  

   - Observed friction points when attempting to create and prepare a custom dataset. The documentation only included a YouTube video link on the home page, which is good a comprehensive guide. But there are still Github issues like the recent [#21847](https://github.com/ultralytics/ultralytics/issues/21847) asking the same question.

   - Users lacked step-by-step instructions in one place for the data pipeline process right from data collection, annotation, dataset configuration, splitting, training, and validation.

2. **GitHub Issues Review:**  
   - Analyzed recent issues labeled like `documentation`, `help wanted`, or `question`. Looked into `Feature` as well just in case there was something being requested as a feature which was already present but not discoverable. So that it can be fixed with better documentation.

   - Found the issues supporting my initial analysis and found new ones as well.

   - This issue [#17811](https://github.com/ultralytics/ultralytics/issues/17811) show the same confusion about dataset configuration.

   - Observed recurring questions regarding augmentations ([#19810](https://github.com/ultralytics/ultralytics/issues/19810), [#21847](https://github.com/ultralytics/ultralytics/issues/21847)), suggesting that existing resources are underutilized due to poor discoverability rather than lack of content. See how issue [#19810](https://github.com/ultralytics/ultralytics/issues/19810) asks for an explanation even though a detailed guide with visual examples already exists.

3. **API Documentation and Linking Review:**  

   - Existing tables of arguments linked to Augmentation Documentation, however it is not the general practice, argument always redirects to source code class with docstring (eg. PyTorch, Tensorflow, etc.), which may cause confusion about augmentation as reflected in GitHub issues.

   - For example, see the Info section at the bottom of train time augmentation args: `For more information about training augmentation operations, see the reference section.` Here `reference section` redirects to API docs. However general practice is to redirect to comprehensive documentation in `Info` section.

## Gap Analysis Findings
Few primary gaps were identified:

1. **Custom Dataset Guidance is Missing:**  
   - There is no dedicated guide for creating and using custom datasets or a one single guide around data pipeline.

   - The home page references a video tutorial without structured instructions, causing user confusion and repetitive support questions.

   - Testing on built-in dataset is a good starting point, for which there are amazing guides/notebooks around it. But Ultralytics being a platform for implementing different YOLO models, people want to try it out on their own data.

   - Users attempting dataset creation encounter friction around annotation formats, dataset configuration files (`dataset.yaml`), splitting strategies, and training validation integration.

    - Data preparation is the first stage of any ML pipeline, and we would want people to move past it quickly and jump right into using the models.

2. **Augmentation Guidance is Hard to Find:**  
   - Existing augmentation resources exist in the documentation but are buried in API references.  

   - Users are not sufficiently guided to these resources in a clear, structured way, leading to repeated issues and forum questions.  

   - The argument linking strategy in tables is not ideal for workflow-oriented learning.

3. **Ultralytics HUB Visibility is Limited**
    - The Ultralytics HUB, the new no-code platform designed to simplify model training and deployment, was not highlighted on the main homepage. As a result, users may miss this resource entirely, despite its potential to accelerate experimentation and adoption of YOLO models.

## Proposed Solution
To address these gaps, the following actions were taken:

1. **Custom Dataset Guide:**  
   - Created a comprehensive guide: **“Create and Prepare Custom Datasets for YOLO”**, covering:  
     - Data collection, annotation, and preprocessing  
     - Dataset structure and configuration (`dataset.yaml`)  
     - Annotation formats and conversion from COCO/Pascal VOC  
     - Quality control and augmentation (built-in, Albumentations, Roboflow)  
     - Dataset splitting (manual and train/validation)  
     - Training with custom datasets (basic and advanced)  
     - Validation strategies  
     - FAQ addressing common dataset concerns

   - Placed this guide prominently on the **home page** right after “New in YOLOv11” and licensing, improving discoverability.

2. **Improved Augmentation Visibility:**  
   - Instead of linking to API-level function documentation in tables, added links to the comprehensive augmentation section in the guide.

   - This directs users to practical workflow guidance without bloating argument tables or overwhelming function references.

   - It would give practical, high-level guidance first, while keeping API references accessible for deeper dives, striking a balance between usability and technical depth.

3. **Ultralytics HUB Highlight:**

    - Added a homepage section for the HUB, making it immediately visible as a no-code platform for training and deploying models.

## Expected Impact
Implementing these changes will:

- **Reduce setup errors and confusion:** Clear dataset guides and augmentation references help users avoid common mistakes like wrong annotation formats, broken dataset.yaml, poor splits. These changes adopt a common best practice to reduce user confusion.

- **Speed up time-to-training:** Users can move from raw data to a working model faster without stalling at the data preparation stage.

- **Improve adoption of YOLOv11:** New users can follow an end-to-end path, from dataset prep to training, directly in the docs, making it easier to try the framework without relying on forums or external sources.

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Adds a comprehensive “YOLO Custom Data Guide” with step-by-step instructions for preparing and training on custom datasets, plus improved docs navigation, cross-links, and minor copy updates to make onboarding smoother and reduce common setup errors. 📚🚀

### 📊 Key Changes
- New guide: docs/en/guides/custom-data.md covering data collection, annotation, YOLO format, dataset.yaml, conversion (COCO/VOC), quality checks, augmentation, splitting, training, and validation.
- Practical examples: code snippets for `convert_coco`, `convert_voc`, `check_dataset`, `autosplit`, and `YOLO().train()` for YOLO11.
- Added resources: embedded tutorial video and “Open in Colab” link to a training notebook.
- Navigation updates: guide listed in Guides index and mkdocs nav; new homepage cards for “Create Custom Dataset for YOLO” and “No-Code Training with HUB”.
- Cross-link improvements: augmentation arguments page now links to relevant API reference pages; train mode page points to the YOLO Data Augmentation Guide.
- Docs polish: minor copy tweaks for clarity; author metadata updated in docs config.

### 🎯 Purpose & Impact
- Easier onboarding: Clear, end-to-end instructions help users prepare datasets and start training YOLO11 faster. ⏱️
- Fewer mistakes: Standardized folder structures, YAML templates, and validation utilities reduce dataset errors. ✅
- Better discoverability: Prominent links/cards guide users to key workflows; consistent cross-links improve navigation and SEO. 🔍
- Practical productivity: Ready-to-use code snippets accelerate dataset conversion, splitting, and augmentation setup. 🧩
- No breaking changes: Documentation-only updates—safe for all users and versions. 🛡️

See the Ultralytics Docs for details.

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @i-akruti, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response — an Ultralytics engineer will review and assist you soon. To ensure a seamless integration of your work, please review the following checklist:

- ✅ Define a Purpose: Clearly explain the purpose of your fix or feature in your PR description, and link to any relevant issues(https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ Synchronize with Source: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ Ensure CI Checks Pass: Verify all Ultralytics Continuous Integration (CI)(https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ Update Documentation: Update the relevant documentation(https://docs.ultralytics.com/) for any new or modified features.
- ✅ Add Tests: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ Sign the CLA: Please ensure you have signed our Contributor License Agreement(https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ Minimize Changes: Limit your changes to the minimum necessary for your bug fix or feature addition. "It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is." — Bruce Lee

For more guidance, please refer to our Contributing Guide(https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

Notes on this PR 💡
- Thanks for the comprehensive documentation improvements and the new guide at docs/en/guides/custom-data.md, plus navigation updates in mkdocs.yml and guides/index.md. The homepage additions highlighting Ultralytics HUB and the guide tile look great.
- Please double-check branding consistency for YOLO11 across the new content and links, and verify that all updated augmentation references in docs/en/macros/augmentation-args.md resolve correctly to the intended API pages.
- If any links reference API or guide anchors, ensure they render as expected after the docs build. See the Ultralytics Docs(https://docs.ultralytics.com) for details.

**Comment by i-akruti** [comment]
I have read the CLA Document and I sign the CLA

**Comment by i-akruti** [comment]
Closing PR due to email linking issue, so not able to sign CLA

**Comment by glenn-jocher** [comment]
Hi @i-akruti—understood. The CLA won’t match if your commits use an unverified email; please verify the address in your GitHub account and ensure your commits use that same email, then amend and force-push (or open a fresh PR) so the bot can re-check; see [Verify your email on GitHub](https://docs.github.com/en/enterprise-cloud@latest/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/verifying-your-email-address) and [Set your commit email address](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-personal-account-on-github/managing-email-preferences/setting-your-commit-email-address).

```bash
git config user.email "your-verified-email@domain.com"
git commit --amend --reset-author --no-edit
git push -f origin <branch>
```
